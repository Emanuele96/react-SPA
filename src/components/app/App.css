* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: Arial, Helvetica, sans-serif;
}

.app {
  width: 100%;
  min-height: 100vh;
  padding: 100px;
}


main {
  height: 100%;
}

.tabs-bar {
  grid-area: tab;
  /*background-color: orange;*/
  height: 100%;
}

.maindisp {
  grid-area: main;
  /*background-color: red;*/
  height: 100%;
}


.sidebar-category {
  grid-area: cat;
  /* background-color: aquamarine;*/
  height: 100%;
}

.grid-container {
  border: black dashed 1px;
  height: 100%;
  width: 100%;
  /*width: 70vw;*/
  display: grid;
  grid-template-areas: 'tab tab tab'
    'main main cat';

  grid-template-rows: 15vh auto;
  grid-template-columns: auto auto 18vw;
  /*background-color: #2196F3;*/
  padding: 0;
  margin: 0;

}

.grid-container>div {
  padding: 0;
  margin: 0;
}

@media only screen and (max-width: 825px) {
  .app {
    padding: 0;
    counter-reset: tabnr;
  }


  /* hides text in the tab buttons*/
  .combination {
    font-size: 0;
  }

  /* displays tab number */
  .combination::after {
    counter-increment: tabnr;
    content: counter(tabnr);
    font-size: 1.5rem;
  }
}


@media only screen and (max-width:825px) and (min-width:376px),
(orientation: landscape) and (max-width:825px) and (min-width:376px) {

  .grid-container {
    grid-template-areas: 'tab''main''cat';
    grid-template-rows: 15vh 85vh auto;
    grid-template-columns: auto;
  }

  .sidebar>h3 {
    display: none;
  }

  .sidebar {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
  }


  form>label {
    padding: 15px;

  }


}

/*bredden til iphone x/8/7/6 = 375px */
@media only screen and (max-width:376px) and (orientation: portrait) {

  .grid-container {
    grid-template-areas: 'tab''main''cat';
    grid-template-rows: 15vh 85vh auto;
    grid-template-columns: auto;
  }

  .sidebar>h3 {
    display: block;
  }

  .sidebar {
    display: inline-block;
    padding-left: 10px;
  }

  form>label {
    padding: 15px;
  }

}