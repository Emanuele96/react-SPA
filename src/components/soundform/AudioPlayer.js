import React from 'react';

import './AudioPlayer.css' 
import piano1 from '../../assets/sounds/piano/1.mp3';
import piano2 from '../../assets/sounds/piano/2.mp3';
import piano3 from '../../assets/sounds/piano/3.mp3';
import piano4 from '../../assets/sounds/piano/4.mp3';
import nature1 from '../../assets/sounds/nature/1.mp3';
import nature2 from '../../assets/sounds/nature/2.mp3';
import nature3 from '../../assets/sounds/nature/3.mp3';
import nature4 from '../../assets/sounds/nature/4.mp3';
import citylife1 from '../../assets/sounds/citylife/1.mp3';
import citylife2 from '../../assets/sounds/citylife/2.mp3';
import citylife3 from '../../assets/sounds/citylife/3.mp3';
import citylife4 from '../../assets/sounds/citylife/4.mp3';


const pianoSongs = [piano1, piano2, piano3, piano4];
const natureSounds = [nature1, nature2, nature3, nature4];
const citylifeSounds = [citylife1, citylife2, citylife3, citylife4];

const AudioPlayer = props => {
  let soundTrack;
  let soundCategory = props.soundCategory;

  if (soundCategory === 'Nature') {
    soundTrack = natureSounds[props.soundTrack - 1];
  } else if (soundCategory === 'Piano') {
    soundTrack = pianoSongs[props.soundTrack - 1];
  } else if (soundCategory === 'City Life') {
    soundTrack = citylifeSounds[props.soundTrack - 1];
  }

  return (
    <div className='sound'>
      <audio
        className='soundPlayer'
        preload='auto'
        src={soundTrack}
        type='audio/mpeg'
        controls
        loop
        autoPlay
      />
    </div>
  );
};

export default AudioPlayer;
