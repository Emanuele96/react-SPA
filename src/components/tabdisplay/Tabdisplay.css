.Tabdisplay {
  border: red dashed 1px;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
}

.combination {
  height: 100px;
  width: 100%;
  font-size: 1.5rem;
  font-weight: 600;
  cursor: pointer;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  transition: transform 0.3s ease-in-out;
}

.combination:hover {
  transform: scale(1.1);
  transition: transform 0.3s ease-in-out;
}

#combination1 {
  background-color: green;
}

#combination2 {
  background-color: palevioletred;
}

#combination3 {
  background-color: powderblue;
}

#combination4 {
  background-color: orange;
}