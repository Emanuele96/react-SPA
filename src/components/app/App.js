import React from 'react';
import Maindisplay from '../maindisplay/Maindisplay';
import Sidebar from '../sidebar/Sidebar';
import Tabdisplay from '../tabdisplay/Tabdisplay';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import './App.css';

class App extends React.Component {
  state = {};

  handleTabClick = e => {
    return '' && e.target.value;
  };

  render() {
    return (
      <div className='App'>
        <header>
          <div>
            <h1>This is header</h1>
          </div>
        </header>
        <main>
          <div className='Box'>
            <div className='InnerBox'>
              <BrowserRouter>
                <Tabdisplay onClick={this.handleTabClick} />
                <Switch>
                  <Route exact path='/' component={Maindisplay} />
                  <Route
                    exact
                    path='/1'
                    render={props => <Maindisplay {...props} number='1' />}
                  />
                  <Route
                    exact
                    path='/2'
                    render={props => <Maindisplay {...props} number='2' />}
                  />
                  <Route
                    exact
                    path='/3'
                    render={props => <Maindisplay {...props} number='3' />}
                  />
                  <Route
                    exact
                    path='/4'
                    render={props => <Maindisplay {...props} number='4' />}
                  />
                </Switch>
              </BrowserRouter>
              <Sidebar />
            </div>
          </div>
        </main>
        <footer>
          <h2>This is footer</h2>
        </footer>
      </div>
    );
  }
}

export default App;
