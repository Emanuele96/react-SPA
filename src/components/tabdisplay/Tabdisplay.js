import React from 'react';
import TabChoice from '../ui/tabchoice/Tabchoice';
import './Tabdisplay.css';

const Tabdisplay = props => {
  return <div className='Tabdisplay'>{createTabChoices(props)}</div>;
};

const createTabChoices = props => {
  let choices = [];
  for (let i = 1; i <= 4; i++) {
    choices.push(<TabChoice tabNumber={i} onClick={props.onClick} value={i} />);
  }

  return choices;
};

export default Tabdisplay;
