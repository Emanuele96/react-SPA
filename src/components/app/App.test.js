import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import Enzyme from 'enzyme';
import { configure } from 'enzyme';
import { shallow, mount, render } from 'enzyme';


import App from './App';
import AudioPlayer from '../soundform/AudioPlayer';

import { conditionalExpression } from '@babel/types';
import Tabdisplay from '../tabdisplay/Tabdisplay';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


describe('Testing app', () => {

  it('renders correctly', () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('Tab click updates app state "selectedTab', () => {

    Enzyme.configure({ adapter: new Adapter() });
    const componentEnzyme = mount(<App />);

    expect(componentEnzyme.state('selectedTab')).toEqual(null);


    let tabButton = componentEnzyme.find('#combination1');
    tabButton.simulate('click');
    expect(componentEnzyme.state('selectedTab')).toEqual("1");

    tabButton = componentEnzyme.find('#combination2');
    tabButton.simulate('click');
    expect(componentEnzyme.state('selectedTab')).toEqual("2");

    tabButton = componentEnzyme.find('#combination3');
    tabButton.simulate('click');
    expect(componentEnzyme.state('selectedTab')).toEqual("3");

    tabButton = componentEnzyme.find('#combination4');
    tabButton.simulate('click');
    expect(componentEnzyme.state('selectedTab')).toEqual("4");

  });

  it('Tab click to update audioplayer', () => {


    Enzyme.configure({ adapter: new Adapter() });

    const componentEnzyme = mount(<App />);
    let audioPlayer = componentEnzyme.find('.soundPlayer');
    expect(audioPlayer.props().src).toEqual(undefined);

    //tab button 1
    let tabButton = componentEnzyme.find('#combination1');
    tabButton.simulate('click');
    audioPlayer = componentEnzyme.find('.soundPlayer');
    expect(audioPlayer.props().src).toEqual("1.mp3");

    //tab button 2
    tabButton = componentEnzyme.find('#combination2');
    tabButton.simulate('click');
    audioPlayer = componentEnzyme.find('.soundPlayer');
    expect(audioPlayer.props().src).toEqual("2.mp3");

    //tab button 3
    tabButton = componentEnzyme.find('#combination3');
    tabButton.simulate('click');
    audioPlayer = componentEnzyme.find('.soundPlayer');
    expect(audioPlayer.props().src).toEqual("3.mp3");

    //tab button 2
    tabButton = componentEnzyme.find('#combination4');
    tabButton.simulate('click');
    audioPlayer = componentEnzyme.find('.soundPlayer');
    expect(audioPlayer.props().src).toEqual("4.mp3");

  });


  it('Categorychange updates appstate "soundCategory', () => {

    Enzyme.configure({ adapter: new Adapter() });

    const componentEnzyme = mount(<App />);

    expect(componentEnzyme.state('soundCategory')).toEqual("Nature");


    // index: 0 = nature, 1 = Piano, 2 = City Life, 
    // 3 = Emosjoner, 4 = Farger, 5 = Dyr
    // 6 = Animals, 7 = Cats, 8 = Rats

    let formButton = componentEnzyme.find('input').at(1);
    formButton.instance().checked = true; 
    formButton.simulate('change');
    
    formButton = componentEnzyme.find('input').at(0);
    expect(formButton.props().checked).toEqual(false);
    formButton = componentEnzyme.find('input').at(1);
    expect(formButton.props().checked).toEqual(true);
    
  }

  );

});