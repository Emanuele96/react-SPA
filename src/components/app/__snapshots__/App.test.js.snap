// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing app renders correctly 1`] = `
<div
  className="app"
>
  <main>
    <div
      className="grid-container"
    >
      <div
        className="tabs-bar"
      >
        <div
          className="tabdisplay"
        >
          <div
            className="tabchoice"
          >
            <button
              className="combination"
              id="combination1"
              onClick={[Function]}
              value={1}
            >
              Kombinasjon 
              1
            </button>
          </div>
          <div
            className="tabchoice"
          >
            <button
              className="combination"
              id="combination2"
              onClick={[Function]}
              value={2}
            >
              Kombinasjon 
              2
            </button>
          </div>
          <div
            className="tabchoice"
          >
            <button
              className="combination"
              id="combination3"
              onClick={[Function]}
              value={3}
            >
              Kombinasjon 
              3
            </button>
          </div>
          <div
            className="tabchoice"
          >
            <button
              className="combination"
              id="combination4"
              onClick={[Function]}
              value={4}
            >
              Kombinasjon 
              4
            </button>
          </div>
        </div>
      </div>
      <div
        className="maindisp"
      >
        <div
          className="maindisplay"
        >
          <div>
            <h2>
              Velkommen! 
            </h2>
            <p>
              Vennligst velg tre kategorier per medietype i vindu til h√∏yre.
            </p>
            <div
              className="sound"
            >
              <audio
                autoPlay={true}
                className="soundPlayer"
                controls={true}
                loop={true}
                preload="auto"
                type="audio/mpeg"
              />
            </div>
          </div>
          <div
            className="favorite-buttons"
          >
            <button
              className="btn delete"
              onClick={[Function]}
            >
              <h3>
                Hent favoritt kombinasjon
              </h3>
            </button>
            <button
              className="btn favorite"
              onClick={[Function]}
            >
              <h3>
                Lagre Kombinasjon
              </h3>
            </button>
          </div>
        </div>
      </div>
      <div
        className="sidebar-category"
      >
        <div
          className="sidebar"
        >
          <h3>
            Kategorier
          </h3>
          <div
            className="choice-form"
          >
            <h3>
              Sound
            </h3>
            <form>
              <label>
                <input
                  checked={true}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Nature"
                />
                Nature
              </label>
              <label>
                <input
                  checked={false}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Piano"
                />
                Piano
              </label>
              <label>
                <input
                  checked={false}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="City Life"
                />
                City Life
              </label>
            </form>
          </div>
          <div
            className="choice-form"
          >
            <h3>
              Tekst
            </h3>
            <form>
              <label>
                <input
                  checked={true}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Emosjoner"
                />
                Emosjoner
              </label>
              <label>
                <input
                  checked={false}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Farger"
                />
                Farger
              </label>
              <label>
                <input
                  checked={false}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Dyr"
                />
                Dyr
              </label>
            </form>
          </div>
          <div
            className="choice-form"
          >
            <h3>
              Bilder
            </h3>
            <form>
              <label>
                <input
                  checked={true}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Animals"
                />
                Animals
              </label>
              <label>
                <input
                  checked={false}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Cats"
                />
                Cats
              </label>
              <label>
                <input
                  checked={false}
                  name="category"
                  onChange={[Function]}
                  type="radio"
                  value="Rats"
                />
                Rats
              </label>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
